Buy/Graph.Client.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  Graph.Client.swift
+//  Buy
+//
+//  Created by Shopify.
+//  Copyright (c) 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+internal typealias JSON = [String : Any]
+
+private struct Header {
+    static var userAgent     = "User-Agent"
+    static var authorization = "Authorization"
+    static var accept        = "Accept"
+    static var contentType   = "Content-Type"
+}
+
+extension Graph {
+    public class Client {
+        
+        public let session:  URLSession
+        
+        private let apiURL:  URL
+        private let headers: [String : String]
+        
+        // ----------------------------------
+        //  MARK: - Init -
+        //
+        public init(shopDomain: String, apiKey: String, session: URLSession = URLSession(configuration: URLSessionConfiguration.default)) {
+            
+            let shopURL  = Client.urlFor(shopDomain)
+            self.apiURL  = Client.urlFor(shopDomain, path: "/api/graphql")
+            self.session = session
+            self.headers = [
```

Comment:
To be consistent with other Buy SDKs, should send:
`X-SDK-Version` which is just version, then `X-SDK-Variant` which is your free form `ios` or `android`